lib/board.rb

```diff
@@ -0,0 +1,57 @@
+class Board
+  attr_accessor :cells
+
+  def initialize
+    @cells = [" ", " ", " ", " ", " ", " ", " ", " ", " "]
+  end
+
+  def reset!
+    self.cells = [" ", " ", " ", " ", " ", " ", " ", " ", " "]
+  end
+
+  def display
+    puts " #{@cells[0]} | #{@cells[1]} | #{@cells[2]} "
+    puts "-----------"
+    puts " #{@cells[3]} | #{@cells[4]} | #{@cells[5]} "
+    puts "-----------"
+    puts " #{@cells[6]} | #{@cells[7]} | #{@cells[8]} "
+  end
+
+  def input_to_index(user_input)
+    user_input.to_i - 1
+  end
+
+  def position(user_input)
+    i = input_to_index(user_input)
+    self.cells[i]
+  end
+
+  def full?
+    !@cells.include?(" ")
+  end
+
+  def turn_count
+    counter = 0
+    self.cells.each do |space|
+        counter += 1 if space == "X" || space == "O"
+    end
+    counter
+  end
+
+  def taken?(position)
+    index  = input_to_index(position)
```

Comment:
We don't really have to create a new variable `index`, instead we can just use `input_to_index(position` in-line.